scripts/optimize.py

```diff
@@ -16,18 +16,30 @@
 -------------
 Given an EnergyDataPackage, this script creates an oemof.solph.EnergySystem and an
 oemof.solph.Model, which is optimized. The EnergySystem with results, meta-results and parameters
-is saved.
+is saved. Further, additional parameters like emission limit are saved in a separate file.
 """
 import os
 import sys
 
-from oemof.solph import EnergySystem, Model
+from oemof.solph import EnergySystem, Model, constraints
 from oemof.outputlib import processing
 
 # DONT REMOVE THIS LINE!
 # pylint: disable=unusedimport
 from oemof.tabular import datapackage  # noqa
 from oemof.tabular.facades import TYPEMAP
+from oemof_b3.tools import data_processing as dp
+
+
+def get_emission_limit():
+    """Reads emission limit from csv file in `preprocessed`."""
+    path_emission_limit = os.path.join(preprocessed, "additional_scalars.csv")
+    scalars = dp.load_b3_scalars(path_emission_limit)
+    emission_scalars = scalars.loc[scalars["carrier"] == "emission"].set_index(
+        "var_name"
+    )
+    emission_limit = emission_scalars.at["emission_limit", "var_value"]
```

Comment:
To help debugging, you could assert that the emission limit is a number.
```suggestion
    emission_limit = emission_scalars.at["emission_limit", "var_value"]
    assert isinstance(emission_limit, (int, float)), "Expected a number for 'emission limit'"
```